Description: Disable kernel pageflipping support.
 There seem to be a number of issues with pageflipping, particularly on
 Gen 3 hardware.
 .
 Some of the required kernel patches which haven't made it upstream are in
 this thread: 
 http://lists.freedesktop.org/archives/intel-gfx/2010-April/006464.html
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=28079
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=27883

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 358baf4..30c9213 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -1458,8 +1458,8 @@ Bool drmmode_pre_init(ScrnInfoPtr scrn, int fd, int cpp)
 				  sizeof(gp));
 	if (has_flipping) {
 		xf86DrvMsg(scrn->scrnIndex, X_INFO,
-			   "Kernel page flipping support detected, enabling\n");
-		intel->use_pageflipping = TRUE;
+			   "Kernel page flipping support detected, but forcibly disabled.\n");
+		intel->use_pageflipping = FALSE;
 		drmmode->flip_count = 0;
 		drmmode->event_context.version = DRM_EVENT_CONTEXT_VERSION;
 		drmmode->event_context.vblank_handler = drmmode_vblank_handler;
diff --git a/src/i830_dri.c b/src/i830_dri.c
index 83d953d..a2b7ec2 100644
--- a/src/i830_dri.c
+++ b/src/i830_dri.c
@@ -1038,7 +1038,7 @@ Bool I830DRI2ScreenInit(ScreenPtr screen)
 
 	info.CopyRegion = I830DRI2CopyRegion;
 #if DRI2INFOREC_VERSION >= 4
-	if (intel->use_pageflipping) {
+	if (intel->use_pageflipping || 1) {
 	    info.version = 4;
 	    info.ScheduleSwap = I830DRI2ScheduleSwap;
 	    info.GetMSC = I830DRI2GetMSC;
